### PR TITLE
Add service discovery calls (QC-176)

### PR DESCRIPTION
**Pull request by Barthelemy**
needs https://github.com/AliceO2Group/QualityControl/pull/145

### CODE REVIEW

**Comment by Barthelemy** [comment]
I will keep only a few commits that are not part of the other PR : 1d5e3ce and 44b01c4.

**Comment by Barthelemy** [comment]
It seems that the problem of not exiting is back. Also note that we have to use a shared pointer in TaskRunner because unique_ptr are incompatible with tasks in dpl. Maybe we could move the servicediscovery object to objectsManager altogether.

**Comment by Barthelemy** [comment]
We must also get the address to consul from the configuration that is itself in Consul ¯\_(ツ)_/¯

**Comment by Barthelemy** [comment]
commits to keep when PR https://github.com/AliceO2Group/QualityControl/pull/162 is merged.
3be5403 
bdfc850 
d9cec32 
620e808

**Comment by Barthelemy** [comment]
I have retested it. It is fine and works. After #162  is merged we can merge this one, not before.

**Comment by awegrzyn** [comment]
@Barthelemy could you let us know when you merge this PR?
https://github.com/AliceO2Group/WebUi/pull/307

**Comment by Barthelemy** [comment]
Sure, There is a complex web of PRs to merge in the right order. I am sorting it out.

File: Framework/include/QualityControl/ObjectsManager.h [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #include "QualityControl/MonitorObject.h"
 #include "QualityControl/Quality.h"
 #include "QualityControl/TaskConfig.h"
+#include "ServiceDiscovery.h"
```
Comment (knopers8) [inline Framework/include/QualityControl/ObjectsManager.h]: Why not "QualityControl/ServiceDiscovery.h", since it is inside QualityControl?

Comment (Barthelemy) [reply Framework/include/QualityControl/ObjectsManager.h]: it was a mistake. Originally the header was not distributed.

**Pull request merged**